include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -866,7 +866,11 @@ namespace internal
 
         for (const auto &i : targets_with_indexset)
           {
-            if (i.first == my_rank)
+            // Skip communication in case we would send to ourselves or when
+            // there are no indices to send (this can still happen in the run
+            // of the consensus algorithms above if the index spaces are
+            // sparse).
+            if (i.first == my_rank || i.second.begin() == i.second.end())
```

Comment:
`i.second` corresponds to an object of type `IndexSet`. Checking for `begin() == end()` is probably more easily understood if you said `i.second.is_empty()`.

Reply:
Separately, I don't think this is sending an empty message. It perhaps sends an empty collection packed up into a string, but that's a non-zero length string.